### PR TITLE
AGENT-283: Remove unwanted ClusterDeployment.Spec fields

### PR DESCRIPTION
**Pull request by pawanpinjarkar**
Remove the unwanted fields from the ClusterDeployment.Spec.

### CODE REVIEW

File: agent/04_agent_configure.sh [hunk 1]
```diff
@@ -122,21 +122,7 @@ metadata:
   namespace: cluster0
 spec:
   baseDomain: ${BASE_DOMAIN}
-  clusterInstallRef:
```
Comment (zaneb) [inline agent/04_agent_configure.sh:125]: This is definitely used: https://github.com/openshift/assisted-service/blob/da8164f04040feae88de132de4cfe02447129dde/internal/controller/controllers/clusterdeployments_controller.go#L140-L146

File: agent/04_agent_configure.sh [hunk 2]
```diff
@@ -122,21 +122,7 @@ metadata:
...
-    group: extensions.hive.openshift.io
-    kind: AgentClusterInstall
-    name: test-agent-cluster-install
-    version: v1beta1
   clusterName: ${CLUSTER_NAME}
-  controlPlaneConfig:
-    servingCertificates: {}
-  platform:
```
Comment (zaneb) [inline agent/04_agent_configure.sh:133]: Platform is used in the Hive controller (not sure how this interacts with the MCE controller?): https://github.com/openshift/hive/blob/master/pkg/controller/clusterdeployment/clusterdeployment_controller.go
I think we should probably set it to `none: {}` until further notice.

Comment (zaneb) [reply agent/04_agent_configure.sh:133]: actually no, let's remove it for now.

**Review by rwsu — APPROVED** [review]
/approve

**Pull request merged**